src/Common/oclBasics.cpp

```diff
@@ -0,0 +1,378 @@
+#include <Common/config.h>
+#if USE_OPENCL
+
+#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#include <malloc.h>
+#endif
+
+#ifdef __APPLE__
+#include <OpenCL/opencl.h>
+#else
+#include <CL/cl.h>
+#endif
+
+#include <algorithm>
+#include <Core/Types.h>
+#include <Common/Exception.h>
+
+#ifndef CL_VERSION_2_0
+#define CL_USE_DEPRECATED_OPENCL_1_2_APIS
+#endif
+
+
+using KernelType = std::remove_reference<decltype(*cl_kernel())>::type;
+
+
+namespace DB
+{
+    namespace ErrorCodes
+    {
+        extern const int OPENCL_ERROR;
+    }
+}
+
+struct OCL
+{
+
+    /**
+     * Structure which represents the most essential settings of common OpenCl entities.
+     */
+    struct Settings
+    {
+        // Platform info
+        cl_uint numberOfPlatformEntries;
```

Comment:
Here and further. Coding style for variables: `numberOfPlatformEntries` -> `number_of_platform_entries`.

Reply:
Renamed all variables ✔️